include/open62541pp/plugin/nodestore.hpp

```diff
@@ -53,25 +53,27 @@ struct ValueBackendDataSource {
      * To use zero-copy reads, set the value of the Variant (DataValue::getValue) without copying,
      * e.g. with Variant::setScalar or Variant::setArray.
      *
+     * @param id The ID of the node whose value was requested
      * @param value The DataValue that is returned to the reader
      * @param range If not empty, then the data source shall return only a selection of the
      *              (nonscalar) data.
      *              Set `UA_STATUSCODE_BADINDEXRANGEINVALID` in `value` if this does not apply
      * @param timestamp Set the source timestamp of `value` if `true`
      * @return StatusCode
      */
-    std::function<StatusCode(DataValue& value, const NumericRange& range, bool timestamp)> read;
+    std::function<StatusCode(NodeId id, DataValue& value, const NumericRange& range, bool timestamp)> read;
```

Comment:
Please pass by `const NodeId&` to avoid copies.

Reply:
Fixed